add warning to project transfer (#10626)
getsentry/sentry-docs: docs/organization/membership/index.mdx

## Transfer a Project

<Note>

> When transferring projects some project data such as releases, and session data are not transferred. Please also note that Organization Settings such as team assignments, and code-mappings will need to be applied to the project again.

Reviewer: ```suggestion
Some project data such as releases and session data won't be transferred during project transfer. In addition, Organization Settings such as team assignments, and code-mappings will need to be specified for the project again. 
```
